homeassistant/helpers/aiohttp_client.py

```diff
@@ -70,8 +70,14 @@ def async_create_clientsession(
         **kwargs,
     )
 
+    async def patched_close() -> None:
+        """Mock close to avoid integrations closing our session."""
```

Comment:
Should we log this? As this would potentially improve the provided library?

Reply:
I added a frame helper and used that here to raise or log.

Reply:
I'll see if I can test the log warning here too.

Reply:
Tests added.